### ISSUE TITLE
Missing dataset code

### ISSUE THREAD

**Issue by HussainAlasmawi** [LMBDA-design/PSAMIL#1]
Dear,

It looks like some code files are missing from the repository.

Specifically, the following imports cannot be found:

from datasets.MILDataset import MILDataset, UNBCBagDataset, get_sampler
from datasets.svhn import MY_SVHN
from datasets.fmnist import MY_FMNIST

Could you please upload these files or let us know where they are located?

Thanks!

**Comment by LMBDA-design** [comment]
Dear Hussain:

Thank you for your interest in our work. We would like to provide some important context regarding the experimental setup and resources. During development, the datasets and evaluation code were adapted from multiple prior research projects, including custom datasets. As a result, the initial code integration was not fully streamlined, and some reported results were generated under separate project frameworks. Additionally, the training process did not employ modern AI engineering practices in certain parts, which may lead to potential technical issues. For example, the logs under Logs/probaligned-psa reflect a workaround solution: due to possible code or memory constraints, continuous 20-epoch training could not be executed; instead, separate training scripts were run for each single epoch.  For fully standardized and controllable training implementations, please utilize the latest libraries such as torchmil.

Under this circumstance, to facilitate verification, we have uploaded the logs and corresponding results obtained during our experiments. You may load and evaluate them using the custom/benchmark lung cancer dataset provided.  The original hosting link was located within China.For broader accessibility, we have just updated the resource to a Google Drive:
https://drive.google.com/drive/folders/1o9o5KT28YgyC1n7Jppk968MWBkNEeQkd?usp=drive_link 

If you intend to reproduce the training, please note that we cannot fully guarantee identical performance. For instance, in the case of the lung cancer dataset, our later attempts did not exceed the performance reported in the paper / uploaded version. Achieving full model performance requires careful coordination of momentum, instance weighting, learning rate, and other hyperparameters. If you plan to run your own training, we encourage careful attention to these factors and extensive experimentation. Typical hyperparameter combinations used in our original runs are visible in the provided logs.

Best regards,
